### PR TITLE
added stdout

### PR DESCRIPTION
**Pull request by cwcai633**
Write stdout to the page

### CODE REVIEW

File: client/services/code_evaluators/PythonCodeRunnerService.js [hunk 1]
```diff
@@ -76,8 +76,9 @@ tie.factory('PythonCodeRunnerService', [
           }
 
           // The run was successful.
+          //outputLines = outputLines.slice(0, outputLines.length / correctnessTestResults.length);
```
Comment (eyurko) [inline client/services/code_evaluators/PythonCodeRunnerService.js]: Don't leave commented-out code in the codebase.

File: client/services/FeedbackGeneratorService.js [hunk 2]
```diff
@@ -92,13 +92,14 @@ tie.factory('FeedbackGeneratorService', [
...
-      correctnessTest, observedOutput) {
+      correctnessTest, observedOutput, stdOutput) {
       var allowedOutputExample = correctnessTest.getAnyAllowedOutput();
       var feedback = FeedbackObjectFactory.create(false);
       feedback.appendTextParagraph('Your code produced the following result:');
       feedback.appendCodeParagraph(
         'Input: ' + _jsToHumanReadable(correctnessTest.getInput()) + '\n' +
-        'Output: ' + _jsToHumanReadable(observedOutput));
+        'Output: ' + _jsToHumanReadable(observedOutput) + '\n' +
+        'stdout: ' + stdOutput);
```
Comment (eyurko) [inline client/services/FeedbackGeneratorService.js]: Why is this lowercase when Input and Output are capitalized?

**Comment by seanlip** [comment]
A reminder: please could you reply to all comments, following the instructions in the playbook? In particular, for comments that you've addressed, please reply "Done." -- this helps keep track of which comments have been addressed and which haven't.

Also, please don't forget to read and respond to @eyurko's comment about the screenshot.

File: client/services/code_evaluators/PythonCodeRunnerService.js (hunk 1 above)
Comment (cwcai633) [inline client/services/code_evaluators/PythonCodeRunnerService.js]: Done.

File: client/services/FeedbackGeneratorService.js (hunk 2 above)
Comment (cwcai633) [inline client/services/FeedbackGeneratorService.js]: Done.

**Comment by cwcai633** [comment]
It works like this.
screenshot 2017-04-08 at 6 05 26 pm
screenshot 2017-04-08 at 6 04 38 pm
screenshot 2017-04-08 at 6 09 16 pm

However, it does not support to print out when there are errors in the code.
screenshot 2017-04-08 at 6 07 27 pm

**Comment by seanlip** [comment]
Thank you for the screenshots! I am passing this to @rabidbit for a UI review. I can give you feedback on the code once the UI is agreed on.

**Comment by rabidbit** [comment]
Please remove the trailing newline after stdout; otherwise, looks fine from a UI perspective Thanks!

File: client/services/FeedbackGeneratorService.js [hunk 3]
```diff
@@ -92,13 +92,14 @@ tie.factory('FeedbackGeneratorService', [
     };
 
     var _getCorrectnessTestFeedback = function(
-      correctnessTest, observedOutput) {
+      correctnessTest, observedOutput, stdOutput) {
```
Comment (seanlip) [inline client/services/FeedbackGeneratorService.js]: Let's just call it stdout. There are two types of "output" here and I think it makes sense to think of the second one as just a stdout feed.

File: client/services/FeedbackGeneratorService.js [hunk 4]
```diff
@@ -183,13 +184,17 @@ tie.factory('FeedbackGeneratorService', [
 
           var correctnessTests = task.getCorrectnessTests();
           var observedOutputs = codeEvalResult.getCorrectnessTestResults();
+          var stdOutputs = codeEvalResult.getOutput();
```
Comment (seanlip) [inline client/services/FeedbackGeneratorService.js]: Why plural?

File: client/services/FeedbackGeneratorService.js [hunk 5]
```diff
@@ -183,13 +184,17 @@ tie.factory('FeedbackGeneratorService', [
...
           for (i = 0; i < correctnessTests.length; i++) {
             var observedOutput = observedOutputs[i];
+            var outputCount = stdOutputs.length / correctnessTests.length;
```
Comment (seanlip) [inline client/services/FeedbackGeneratorService.js]: It's not always going to be the case that the output for each test case is of equal length. You might need to find a way to catch each of them somewhere.

File: client/services/code_evaluators/PythonCodeRunnerService.js [hunk 6]
```diff
@@ -77,7 +77,7 @@ tie.factory('PythonCodeRunnerService', [
 
           // The run was successful.
           return CodeEvalResultObjectFactory.create(
-            code, outputLines.join('\n'), correctnessTestResults,
+            code, outputLines, correctnessTestResults,
```
Comment (seanlip) [inline client/services/code_evaluators/PythonCodeRunnerService.js]: If we are doing this then perhaps we might need to change the output property in CodeEvalResult to outputLines.

**Comment by rabidbit** [comment]
Re: question (1) above, you could make the same argument for syntax highlighting, errors, and whether or not the code passes actual tests. I'd say not being able to print to console could be part of the "whiteboard" mode "where you are not provided with syntax highlighting and errors, no ability to print to console, and there is a timer". We ask if users would find this useful in the user study. We can also encourage people to try in whiteboard mode if/when we implement that. So, I think it's probably OK to let users print in the current UI (non-whiteboard mode).

File: client/services/FeedbackGeneratorService.js (hunk 5 above)
Comment (cwcai633) [inline client/services/FeedbackGeneratorService.js]: TODO: support variable length stdout.

File: client/services/FeedbackGeneratorService.js (hunk 3 above)
Comment (cwcai633) [inline client/services/FeedbackGeneratorService.js]: Done.

File: client/services/FeedbackGeneratorService.js (hunk 4 above)
Comment (cwcai633) [inline client/services/FeedbackGeneratorService.js]: Done.

File: client/services/code_evaluators/PythonCodeRunnerService.js (hunk 6 above)
Comment (cwcai633) [inline client/services/code_evaluators/PythonCodeRunnerService.js]: Done.

**Comment by rabidbit** [comment]
Per offline discussion, this feature will be placed on hold and not merged into TIE. We will revisit this if we decide to incorporate it later on.

**Comment by seanlip** [comment]
Closing, since we are not going to merge this in right now (per previous discussion).

**Pull request closed without merging**